base/intfuncs.jl

```diff
@@ -159,12 +159,16 @@ end
 invmod(n::Integer, m::Integer) = invmod(promote(n,m)...)
 
 # ^ for any x supporting *
-to_power_type(x::Number) = oftype(x*x, x)
-to_power_type(x) = x
-@noinline throw_domerr_powbysq(p) = throw(DomainError(p,
+to_power_type(x) = convert(Base.promote_op(*, typeof(x), typeof(x)), x)
+@noinline throw_domerr_powbysq(::Any, p) = throw(DomainError(p,
     string("Cannot raise an integer x to a negative power ", p, '.',
```

Comment:
Since this is generic function, it might make sense to make this message generic if it doesn't make it (much) less informative. What about just writing "Cannot raise an integer x to a negative power. Try converting input to float". We could keep this version but restrict the inputs to `Integer`s.